Editorial review: Document anchor-scope property (#42227)
mdn/content: files/en-us/web/css/guides/anchor_positioning/using/index.md

}
```

## Anchor scoping

> When multiple anchor elements are given the same {{cssxref("anchor-name")}} value and a positioned element is associated with that anchor name (by specifying the name as its {{cssxref("position-anchor")}} property value), the positioned element will be associated with the _last_ anchor element in the source order with that anchor name.

Reviewer: ```suggestion
When multiple anchor elements are given the same {{cssxref("anchor-name")}} value and a positioned element has that name as its {{cssxref("position-anchor")}} property value, the positioned element will be associated with the _last_ anchor element in the source order with that `anchor-name` value.
```
the element is associated with another element, not with the name.
not 100% sure if all the elements have that name, or if really only the last one has a name. the last edit makes finding the answer unnecessary.

For example, if a document contains multiple repeated components, each with a positioned element tethered to an anchor, all the positioned elements will be anchored to the last anchor on the page unless each component uses a different anchor name. This is likely not the desired behavior.

> The {{cssxref("anchor-scope")}} property can fix this problem by making it so that each positioned element can only be anchored to an element within the same subtree of the element that has the scope set on it:

Reviewer: how about something like:

```suggestion
With the {{cssxref("anchor-scope")}} property, you can limit the visibility, or "scope', of an `anchor-name` value to a specific subtree, which enables the association of multiple same-named anchors with their own positioned elements within a single document:
```

- `anchor-scope: all` sets the scope so that _any_ `anchor-name` values set in the subtree can only be bound to by positioned elements in the same subtree.

> - `anchor-scope: --my-anchor` sets the scope so that the specified `anchor-name` value, when set in the subtree, can only be bound to by positioned elements in the same subtree.

Reviewer: suggest mentioning `none` is the default, and somehow alluding to the fact that it can be a comma separated list.

Author: Good call. I've updated the second bullet to indicate that multiple comma-separated values can be set, and added a third bullet to explain the `none` value.

> For example, let's say you have an anchor and a anchor-positioned element inside a scoped container, and another positioned element outside the container:

Author: OK, I've rewritten the example to include three sets.

```html live-sample___anchor-scope
<section class="scoped">
  <div class="anchor">⚓︎</div>
  <div class="positioned">Positioned 1</div>
</section>

<div class="positioned">Positioned 2</div>
```

We scope the container using `anchor-scope: --my-anchor`:

```css hidden live-sample___anchor-scope
.scoped {
  padding: 20px;
  background: #eee;
}

.anchor {
  font-size: 1.8rem;
  color: white;
  text-shadow: 1px 1px 1px black;
  background-color: blue;
  width: fit-content;
  padding: 3px;
}

.positioned {
  background: orange;
  width: fit-content;
  padding: 3px;

>   position: absolute;

Reviewer: end the hidden part here (i can't add closing tick marks in this editor)

```suggestion
```css live-sample___anchor-scope
li {
  anchor-scope: --my-anchor;
}
.positioned {
  position: absolute;
```

  position-anchor: --my-anchor;
  position-area: right;
}

.anchor {
  anchor-name: --my-anchor;
}
```

```css live-sample___anchor-scope
.scoped {
  anchor-scope: --my-anchor;
}

> ```

Reviewer: ```suggestion
```
moved up
